src/main/java/com/google/gwt/site/webapp/client/GWTProjectEntryPoint.java

```diff
@@ -73,117 +105,139 @@ public void onModuleLoad() {
+    forEach("#submenu a[href='" + path + "']", el -> {
+      if (!el.style.display.equals("none")) {
+        container[0] = el;
+      }
+    });
+    HTMLElement selectedItem = container[0];
+    String mainNavigationHref;
+    String mainTitle;
+    if (selectedItem != null) {
+
+      showBranch(selectedItem);
+
+      List<HTMLElement> liParents = parentMenuItems(selectedItem);
 
-    showBranch(selectedItem);
+      HTMLElement subMenuItem = liParents.get(liParents.size() - 1);
 
-    GQuery liParents = selectedItem
-        .parentsUntil("#submenu")
-        .filter("li");
-    GQuery subMenuItem = liParents.last();
+      subMenuItem.style.display = "";
+      forEach("#submenu .active", el -> {
+        if (!liParents.contains(el)) {
+          el.classList.remove("active");
+        }
+      });
+      liParents.add(selectedItem);
+      liParents.forEach(el -> {
+        if (el != selectedItem.parentElement) {
+          el.classList.add("active");
+        }
+      });
+      mainNavigationHref = subMenuItem.querySelector("a").getAttribute("href");
+      mainTitle = subMenuItem.querySelector("a").textContent + " - " + selectedItem.textContent;
+    } else {
+      mainNavigationHref = Window.Location.getPath();
+      mainTitle = isOverviewPage(mainNavigationHref) ? "Overview" : "Project";
+    }
 
-    subMenuItem.show();
-    String mainNavigationHref = subMenuItem.children("a").attr("href");
+    forEach("#nav a.active", el -> el.classList.remove("active"));
+    forEach("#nav a[href='" + mainNavigationHref + "']", el -> el.classList.add("active"));
 
-    $("#nav a.active").removeClass("active");
-    $("#nav a[href='" + mainNavigationHref + "']").addClass("active");
-    $("#submenu .active").not(liParents).removeClass("active");
-    liParents.add(selectedItem).not(selectedItem.parent()).addClass("active");
 
     // Change the page title for easy bookmarking
-    $("title").text("[GWT] " + subMenuItem.children("a").text() + " - " + selectedItem.text());
+    querySelector("title").textContent = "[GWT] " + mainTitle;
 
     boolean homePage = isHomePage(path);
     boolean overviewPage = isOverviewPage(path);
-    $("#nav").toggleClass("alwaysOpen", homePage);
-    $("#content").toggleClass("home", homePage);
-    if (homePage || overviewPage) {
-      $("#submenu").hide();
-    } else {
-      $("#submenu").show();
+    querySelector("#nav").classList.toggle("alwaysOpen", homePage);
+    querySelector("#content").classList.toggle("home", homePage);
+    if (querySelector("#submenu") != null) {
+      if (homePage || overviewPage) {
+        querySelector("#submenu").style.display = "none";
+      } else {
+        querySelector("#submenu").style.display = "";
+      }
     }
 
     maybeStyleHomepage();
   }
 
+  private List<HTMLElement> parentMenuItems(HTMLElement selectedItem) {
+    ArrayList<HTMLElement> ret = new ArrayList<>();
+    elemental2.dom.Element current = selectedItem;
```

Comment:
```suggestion
    Element current = selectedItem;
```
and add an import 

(also line 253)